### PR TITLE
Docs Plan Execution

### PR DESCRIPTION
**Pull request by codeboy5**
Documents the plans execute function implemented in the PR #68 .
Issue #72

### CODE REVIEW

File: syftlib/src/main/java/org/openmined/syft/execution/Plan.kt [hunk 1]
```diff
@@ -16,6 +16,8 @@ private const val TAG = "syft.processes.Plan"
 class Plan(val planId: String) {
     private var pytorchModule: Module? = null
 
+    // The execute function takes in the model parameters, training data and client config as input parameters.
```
Comment (vkkhare) [inline syftlib/src/main/java/org/openmined/syft/execution/Plan.kt:19]: This is a public api, so I would prefer if you write complete java docs for this class. 

In essence I would like it to have the following:
- description of the class in the beginning
- description of constructor with it's arguments
- what is pytorchModule
- a java doc for execute method
- java doc for generate script

File: syftlib/src/main/java/org/openmined/syft/execution/Plan.kt [hunk 2]
```diff
@@ -35,12 +37,17 @@ class Plan(val planId: String) {
...
         val batchSize = IValue.from(
             Tensor.fromBlob(longArrayOf(clientConfig.batchSize), longArrayOf(1))
         )
+
+        // lr is the pytorch IValue tensor containing the learning rate specificed in the client configs.
         val lr = IValue.from(
             Tensor.fromBlob(floatArrayOf(clientConfig.lr), longArrayOf(1))
         )
+
+        // We feed in the training data, output labels, batchSize, learning rate and other parameters to the forward function of the pytorch module.
```
Comment (vkkhare) [inline syftlib/src/main/java/org/openmined/syft/execution/Plan.kt:50]: May be break the comment into two lines. Also mention the output of the forward.

File: syftlib/src/main/java/org/openmined/syft/execution/Plan.kt (hunk 1 above)
Comment (vkkhare) [reply syftlib/src/main/java/org/openmined/syft/execution/Plan.kt:19]: Also for these methods describe the output structure. For reference, you can check pytorch java docs

File: syftlib/src/main/java/org/openmined/syft/execution/Plan.kt [hunk 3]
```diff
@@ -12,10 +12,24 @@ import java.io.FileOutputStream
...
+ * tensor after doing some computation.
+ */
 @ExperimentalUnsignedTypes
 class Plan(val planId: String) {
     private var pytorchModule: Module? = null
 
+    /**
+     * Loads a serialized TorchScript module from the specified path on the disk.
+     *
+     * @param model The model parameters.
```
Comment (vkkhare) [inline syftlib/src/main/java/org/openmined/syft/execution/Plan.kt:28]: Model hosting model parameters would be better I guess

File: syftlib/src/main/java/org/openmined/syft/execution/Plan.kt [hunk 4]
```diff
@@ -12,10 +12,24 @@ import java.io.FileOutputStream
...
+     * @param trainingBatch The training data.
```
Comment (vkkhare) [inline syftlib/src/main/java/org/openmined/syft/execution/Plan.kt:29]: Training batch = Pair<IValue,IValue>

File: syftlib/src/main/java/org/openmined/syft/execution/Plan.kt [hunk 5]
```diff
@@ -12,10 +12,24 @@ import java.io.FileOutputStream
...
+     * @param clientConfig The hyperparamters for the model.
+     * @return The output of passing the training data through the pytorch Module.
```
Comment (vkkhare) [inline syftlib/src/main/java/org/openmined/syft/execution/Plan.kt:31]: The output contains the loss, accuracy values as defined while creating plan. But it also contains the updated parameters of the model. These parameters are then saved manually by user

File: syftlib/src/main/java/org/openmined/syft/execution/Plan.kt (hunk 4 above)
Comment (codeboy5) [reply syftlib/src/main/java/org/openmined/syft/execution/Plan.kt:29]: Should I add that it is the training data along with the labels ?

Comment (vkkhare) [reply syftlib/src/main/java/org/openmined/syft/execution/Plan.kt:29]: Yeah that can also be added. The syntactic description is also needed like it is a pair of ivalues of labels and training data

Comment (codeboy5) [reply syftlib/src/main/java/org/openmined/syft/execution/Plan.kt:29]: Ok

Comment (codeboy5) [reply syftlib/src/main/java/org/openmined/syft/execution/Plan.kt]: Made the changes

File: syftlib/src/main/java/org/openmined/syft/execution/Plan.kt [hunk 6]
```diff
@@ -35,17 +50,28 @@ class Plan(val planId: String) {
...
         )
+
+        // We feed in the training data to the forward function of the pytorchModule.
         return localModuleState.forward(x, y, batchSize, lr, *params)
     }
 
     // TODO The way a plan is generated should be provided.
     // TODO We should enforce this to  happen in a background thread.
+    /**
+     * Loads a serialized TorchScript module from the specified path on the disk.
```
Comment (vkkhare) [inline syftlib/src/main/java/org/openmined/syft/execution/Plan.kt:70]: Remove serialised in this line

File: syftlib/src/main/java/org/openmined/syft/execution/Plan.kt [hunk 7]
```diff
@@ -55,6 +81,13 @@ class Plan(val planId: String) {
         pytorchModule = Module.load(torchscriptLocation)
     }
 
+    /**
+     * Writes the module to the torchscript and returns the absolute path.
+     *
+     * @param filesDir directory where the .pt file is present, on which the TorchScript model is saved.
```
Comment (vkkhare) [inline syftlib/src/main/java/org/openmined/syft/execution/Plan.kt:87]: filesDir is the directory where the torchscript file will be written

File: syftlib/src/main/java/org/openmined/syft/execution/Plan.kt [hunk 8]
```diff
@@ -55,6 +81,13 @@ class Plan(val planId: String) {
...
+     * @param obj Stream containing the contents of the TorchScript module.
```
Comment (vkkhare) [inline syftlib/src/main/java/org/openmined/syft/execution/Plan.kt:88]: Protobuf stream

File: syftlib/src/main/java/org/openmined/syft/execution/Plan.kt [hunk 9]
```diff
@@ -12,10 +12,25 @@ import java.io.FileOutputStream
...
 class Plan(val planId: String) {
     private var pytorchModule: Module? = null
 
+    /**
+     * Loads a serialized TorchScript module from the specified path on the disk.
+     *
+     * @param model Model hosting model parameters.
+     * @param trainingBatch Training batch = Pair<IValue,IValue>. Contains the training data along with the labels.
+     * @param clientConfig The hyperparamters for the model.
+     * @return The output contains the loss, accuracy values as defined while creating plan. But it also
```
Comment (vkkhare) [inline syftlib/src/main/java/org/openmined/syft/execution/Plan.kt:31]: 'but' not needed here

**Comment by vkkhare** [comment]
Also rebase the branch

**Comment by vkkhare** [comment]
@all-contributors please add @codeboy5  for documentation
